vcr/holder/openid.go

```diff
@@ -149,18 +163,18 @@ func (h openidHandler) HandleCredentialOffer(ctx context.Context, offer oidc4vci
 	retrieveCtx := audit.Context(ctx, "app-openid4vci", "VCR/OpenID4VCI", "RetrieveCredential")
 	retrieveCtx, cancel := context.WithTimeout(retrieveCtx, h.config.Timeout)
 	defer cancel()
-	credential, err := h.retrieveCredential(retrieveCtx, issuerClient, offer, accessTokenResponse)
+	credential, err := h.retrieveCredential(retrieveCtx, issuerClient, offeredCredential.CredentialDefinition, accessTokenResponse)
 	if err != nil {
 		return oidc4vci.Error{
 			Err:        fmt.Errorf("unable to retrieve credential: %w", err),
 			Code:       oidc4vci.ServerError,
 			StatusCode: http.StatusInternalServerError,
 		}
 	}
-	if err = credentialTypesMatchOffer(h.jsonldReader, *credential, offer.Credentials[0]); err != nil {
+	if err = oidc4vci.ValidateDefinitionWithCredential(*credential, *offeredCredential.CredentialDefinition); err != nil {
 		return oidc4vci.Error{
 			Err:        fmt.Errorf("received credential does not match offer: %w", err),
-			Code:       oidc4vci.UnsupportedCredentialType,
+			Code:       oidc4vci.InvalidRequest,
```

Comment:
change is correct?

Reply:
spec doesn't define a response to the credential offer, so its whatever we define to be correct. IMO it is correct.